src/main/java/duke/model/DukePP.java

```diff
@@ -0,0 +1,78 @@
+package duke.model;
+
+import duke.commons.LogsCenter;
+import duke.exception.DukeException;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import java.util.logging.Logger;
+
+/**
+ * Wraps all memory data of Duke++
+ * Implements the interface of model module.
+ */
+
+public class DukePP implements Model {
```

Comment:
This is an interesting way to structure your code. It looks like this wraps around every operation duke++ is intended to do. I'm concerned that this may violate the separation of concerns and single responsibility principles.  

All related logic must be grouped into a single class. Adding, deleting, updating an expense must belong to one class. Stuff like monthly income list could belong to another class. You may want to think about this.